File: Statista/Statista.Domain/Entities/Survey/SurveyConfiguration.cs
namespace Statista.Domain.Entities;

public class SurveyConfiguration
{
    public Guid Id { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public bool IsAnonymous { get; set; } = false;
    public Guid SurveyId { get; set; }
    public Survey Survey { get; set; }
}

File: statistika_mobile/lib/features/survey/domain/model/survey_configuration/survey_configuration.dart
import 'package:json_annotation/json_annotation.dart';

part 'survey_configuration.g.dart';

@JsonSerializable()
class SurveyConfiguration {
  SurveyConfiguration({
    required this.id,
    required this.startDate,
    required this.endDate,
    required this.isAnonymous,
    required this.surveyId,
  });

  final String id;
  final DateTime? startDate;
  final DateTime? endDate;
  final bool isAnonymous;
  final String surveyId;

  factory SurveyConfiguration.fromJson(Map<String, dynamic> json) =>
      _$SurveyConfigurationFromJson(json);

  Map<String, dynamic> toJson() => _$SurveyConfigurationToJson(this);
}
